src/modules/core/components/Icon/Icon.tsx

```diff
@@ -75,6 +75,27 @@ const Icon = ({
   const iconHref = typeof icon === 'object' ? `#${icon.default.id}` : icon;
   const iconTitle =
     typeof title === 'object' ? formatMessage(title, titleValues) : title;
+  const getViewBox = useCallback(() => {
+    switch (appearance.size) {
+      case 'extraTiny':
+        return '0 0 12 12';
+      case 'tiny':
+        return '0 0 14 14';
+      case 'small':
+        return '0 0 16 16';
+      case 'normal':
+        return '0 0 18 18';
+      case 'medium':
+        return '0 0 28 28';
+      case 'large':
+        return '0 0 42 42';
+      case 'huge':
+        return '0 0 64 64';
+      default:
+        return '0 0 18 18';
+    }
+  }, [appearance]);
```

Comment:
Why you "can" programatically change the viewbox of svg elements, it doesn't make the actual `path` render better at different sizes -- that's why I suggested we need **different version** of file sizes

Also, the same effect can be achieved by setting a width and height for the icon component in css, wihtout involving js code _(but again, it will not look pixel-perfect for all sizes)_